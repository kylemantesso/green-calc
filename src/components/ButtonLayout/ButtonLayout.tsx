import * as React from "react";
import styled from "styled-components/native";
import { Button, ButtonLabelType, OPERATORS } from "../Buttton/Button";
import { Column } from "../Column/Column";

const Row = styled.View`
  flex-direction: row;
  margin-bottom: 24px;
`;

interface Props {
  handler: (label: ButtonLabelType) => void;
  lastButton?: ButtonLabelType;
}

const ButtonRow = ({
  handler,
  lastButton,
  labels
}: Props & { labels: ButtonLabelType[] }) => (
  <Row>
    {labels.map((label, ix) => (
      <Button
        key={`${label}-${ix}`}
        handler={handler}
        label={label}
        selected={lastButton !== "=" && lastButton === label && OPERATORS.includes(label)}
      />
    ))}
  </Row>
);

export const ButtonLayout = ({ handler, lastButton }: Props) => {
  return (
    <Column>
      <ButtonRow
        handler={handler}
        lastButton={lastButton}
        labels={["C", "+", "-", "X"]}
      />
      <ButtonRow
        handler={handler}
        lastButton={lastButton}
        labels={["0", "1", "2", "/"]}
      />
      <ButtonRow
        handler={handler}
        lastButton={lastButton}
        labels={["3", "4", "5", "="]}
      />
    </Column>
  );
};
