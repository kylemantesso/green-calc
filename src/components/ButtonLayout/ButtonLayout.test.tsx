import React from "react";

import renderer from "react-test-renderer";
import { ButtonLayout } from "./ButtonLayout";

test("renders correctly", () => {
  const handler = jest.fn();
  const tree = renderer.create(<ButtonLayout handler={handler} />).toJSON();
  expect(tree).toMatchSnapshot();
});
