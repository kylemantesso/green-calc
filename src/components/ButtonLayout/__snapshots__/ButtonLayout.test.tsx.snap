// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexDirection": "column",
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "flex-end",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "marginBottom": 24,
        },
      ]
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFF",
          "borderColor": "#27aa80",
          "borderRadius": 10,
          "borderStyle": "solid",
          "borderWidth": 5,
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "marginBottom": 4,
          "marginLeft": 4,
          "marginRight": 4,
          "marginTop": 4,
          "opacity": 1,
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 10,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#333",
              "fontSize": 38,
            },
          ]
        }
      >
        C
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFF",
          "borderColor": "#27aa80",
          "borderRadius": 10,
          "borderStyle": "solid",
          "borderWidth": 5,
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "marginBottom": 4,
          "marginLeft": 4,
          "marginRight": 4,
          "marginTop": 4,
          "opacity": 1,
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 10,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#333",
              "fontSize": 38,
            },
          ]
        }
      >
        +
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFF",
          "borderColor": "#27aa80",
          "borderRadius": 10,
          "borderStyle": "solid",
          "borderWidth": 5,
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "marginBottom": 4,
          "marginLeft": 4,
          "marginRight": 4,
          "marginTop": 4,
          "opacity": 1,
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 10,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#333",
              "fontSize": 38,
            },
          ]
        }
      >
        -
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFF",
          "borderColor": "#27aa80",
          "borderRadius": 10,
          "borderStyle": "solid",
          "borderWidth": 5,
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "marginBottom": 4,
          "marginLeft": 4,
          "marginRight": 4,
          "marginTop": 4,
          "opacity": 1,
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 10,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#333",
              "fontSize": 38,
            },
          ]
        }
      >
        X
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "marginBottom": 24,
        },
      ]
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFF",
          "borderColor": "#27aa80",
          "borderRadius": 10,
          "borderStyle": "solid",
          "borderWidth": 5,
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "marginBottom": 4,
          "marginLeft": 4,
          "marginRight": 4,
          "marginTop": 4,
          "opacity": 1,
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 10,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#333",
              "fontSize": 38,
            },
          ]
        }
      >
        0
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFF",
          "borderColor": "#27aa80",
          "borderRadius": 10,
          "borderStyle": "solid",
          "borderWidth": 5,
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "marginBottom": 4,
          "marginLeft": 4,
          "marginRight": 4,
          "marginTop": 4,
          "opacity": 1,
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 10,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#333",
              "fontSize": 38,
            },
          ]
        }
      >
        1
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFF",
          "borderColor": "#27aa80",
          "borderRadius": 10,
          "borderStyle": "solid",
          "borderWidth": 5,
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "marginBottom": 4,
          "marginLeft": 4,
          "marginRight": 4,
          "marginTop": 4,
          "opacity": 1,
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 10,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#333",
              "fontSize": 38,
            },
          ]
        }
      >
        2
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFF",
          "borderColor": "#27aa80",
          "borderRadius": 10,
          "borderStyle": "solid",
          "borderWidth": 5,
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "marginBottom": 4,
          "marginLeft": 4,
          "marginRight": 4,
          "marginTop": 4,
          "opacity": 1,
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 10,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#333",
              "fontSize": 38,
            },
          ]
        }
      >
        /
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "marginBottom": 24,
        },
      ]
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFF",
          "borderColor": "#27aa80",
          "borderRadius": 10,
          "borderStyle": "solid",
          "borderWidth": 5,
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "marginBottom": 4,
          "marginLeft": 4,
          "marginRight": 4,
          "marginTop": 4,
          "opacity": 1,
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 10,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#333",
              "fontSize": 38,
            },
          ]
        }
      >
        3
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFF",
          "borderColor": "#27aa80",
          "borderRadius": 10,
          "borderStyle": "solid",
          "borderWidth": 5,
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "marginBottom": 4,
          "marginLeft": 4,
          "marginRight": 4,
          "marginTop": 4,
          "opacity": 1,
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 10,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#333",
              "fontSize": 38,
            },
          ]
        }
      >
        4
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFF",
          "borderColor": "#27aa80",
          "borderRadius": 10,
          "borderStyle": "solid",
          "borderWidth": 5,
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "marginBottom": 4,
          "marginLeft": 4,
          "marginRight": 4,
          "marginTop": 4,
          "opacity": 1,
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 10,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#333",
              "fontSize": 38,
            },
          ]
        }
      >
        5
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFF",
          "borderColor": "#27aa80",
          "borderRadius": 10,
          "borderStyle": "solid",
          "borderWidth": 5,
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "marginBottom": 4,
          "marginLeft": 4,
          "marginRight": 4,
          "marginTop": 4,
          "opacity": 1,
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 10,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#333",
              "fontSize": 38,
            },
          ]
        }
      >
        =
      </Text>
    </View>
  </View>
</View>
`;
